Plugins/Public/base_plugin/PlayerBase.cpp

```diff
@@ -394,14 +416,30 @@ void PlayerBase::Save()
 		fprintf(file, "shielddmgtaken = %f\n", damage_taken_since_last_threshold);
 
 		fprintf(file, "money = %I64d\n", money);
-		fprintf(file, "system = %u\n", system);
+		auto sysInfo = Universe::get_system(system);
+		fprintf(file, "system = %s\n", sysInfo->nickname);
 		fprintf(file, "pos = %0.0f, %0.0f, %0.0f\n", position.x, position.y, position.z);
 
-		fprintf(file, "destsystem = %u\n", destsystem);
-		fprintf(file, "destposition = %0.0f, %0.0f, %0.0f\n", destposition.x, destposition.y, destposition.z);
-
 		Vector vRot = MatrixToEuler(rotation);
 		fprintf(file, "rot = %0.0f, %0.0f, %0.0f\n", vRot.x, vRot.y, vRot.z);
+		if (mapArchs[basetype].ishubreturn)
+		{
+			const auto& systemInfo = Universe::get_system(destSystem);
+			fprintf(file, "destsystem = %s\n", systemInfo->nickname);
+
+			fprintf(file, "destpos = %0.0f, %0.0f, %0.0f\n", destPos.x, destPos.y, destPos.z);
+
+			Vector destRot = MatrixToEuler(destOri);
+			fprintf(file, "destori = %0.0f, %0.0f, %0.0f\n", destRot.x, destRot.y, destRot.z);
+		}
+		else if (mapArchs[basetype].isjump && destObject && pub::SpaceObj::ExistsAndAlive(destObject) == 0)
```

Comment:
ExistsAndAlive return code weirdness again - assume this is an enum somewhere?

Reply:
Again, it returns 0 for alive, -2 for dead. I tried to make it return an enum internally but that wasn't an option. In hindsight I should just make a wrapper function interpreting the value and returning appropriate boolean.